web/src/demo_components/list_group_item.dart

```diff
@@ -0,0 +1,241 @@
+  ///
+  /// Default: false
+  bool isActive;
+
+  /// Whether the [ListGroupItem] is disabled.
+  ///
+  /// See: <http://v4-alpha.getbootstrap.com/components/list-group/#disabled-items>
+  ///
+  /// Default: false
+  @Accessor(key: 'disabled', keyNamespace: '')
+  bool isDisabled;
+
+  /// The HTML `href` attribute value for the [ListGroupItem].
+  ///
+  /// If set, the item will render via [Dom.a].
+  ///
+  /// _Proxies [DomProps.href]_
+  @Accessor(keyNamespace: '')
+  String href;
+
+  /// The HTML `target` attribute value for the [ListGroupItem].
+  ///
+  /// If set, the item will render via [Dom.a].
+  ///
+  /// _Proxies [DomProps.target]_
+  @Accessor(keyNamespace: '')
+  String target;
+
+  /// The HTML `type` attribute value for the [ListGroupItem] when
+  /// rendered via [Dom.button].
+  ///
+  /// This will only be applied if [onClick] is also set.
+  ///
+  /// _Proxies [DomProps.type]_
+  ///
+  /// Default: [ButtonType.BUTTON]
+  ButtonType type;
+}
+
+@Component()
+class ListGroupItemComponent extends UiComponent<ListGroupItemProps> {
+  @override
+  Map getDefaultProps() => (newProps()
+    ..elementType = ListGroupItemElementType.SPAN
+    ..skin = ListGroupItemSkin.DEFAULT
+    ..isActive = false
+    ..isDisabled = false
+    ..type = ButtonType.BUTTON
+    ..headerSize = ListGroupItemHeaderElementSize.H5
+  );
+
+  @override
+  render() {
+    var children = props.children;
+
+    if (props.header != null) {
+      children = [
+        renderItemHeader(),
+        (Dom.p()
+          ..className = 'list-group-item-text'
+          ..key = 'item-text'
+        )(props.children)
+      ];
+    }
+
+    BuilderOnlyUiFactory<DomProps> factory = _getItemDomNodeFactory();
+
+    return (factory()
+      ..addProps(copyUnconsumedDomProps())
+      ..className = _getItemClasses().toClassName()
+      ..href = props.href
+      ..target = props.target
+      ..type = _isActionItem ? props.type.typeName : null
+      ..disabled = _useDisabledAttr ? props.isDisabled : null
+      ..addProps(ariaProps()
+        ..disabled = !_useDisabledAttr ? props.isDisabled : null
+      )
+    )(children);
+  }
+
+  ReactElement renderItemHeader() {
+    if (props.header == null) return null;
+
+    var headerClasses = new ClassNameBuilder.fromProps(props.headerProps)
+      ..add('list-group-item-heading');
+
+    return (props.headerSize.componentBuilderFactory()
+      ..addProps(props.headerProps)
+      ..className = headerClasses.toClassName()
+      ..key = 'item-header'
+    )(props.header);
+  }
+
+  BuilderOnlyUiFactory<DomProps> _getItemDomNodeFactory() {
+    var factory;
+
+    if (props.href != null) {
+      factory = Dom.a;
+    } else if (props.onClick != null) {
+      factory = Dom.button;
+    } else {
+      factory = props.elementType.componentBuilderFactory;
+    }
+
+    return factory;
+  }
+
+  ClassNameBuilder _getItemClasses() {
```

Comment:
@jacehensley-wf it's just an architectural thing that I've started leaning towards since consumers of WSD have been having difficulties extending components to build slight variations.

Placing the classes of a root node in a method on the component allows the consumer to then build something slightly different, simply by overriding, instead of having to *nest* a component within another component (which is less performant).

To make it truly worthwhile - components should also be set up like this - so that, for example, building a slightly different button (like a stateful toggle button) - can be done more easily:

```dart
@Component()
class ButtonComponent<T extends ButtonProps, S extends UiState> 
    extends UiStatefulComponent<T, S> {
  // ...

  @override
  render() {
    return renderButton(props.children);
  }

  ReactElement renderButton(dynamic children) {
    BuilderOnlyUiFactory<DomProps> factory = _buttonDomNodeFactory;

    return (factory()
      ..addProps(copyUnconsumedDomProps())
      ..className = getButtonClasses().toClassName()
      ..href = props.href
      ..target = props.target
      ..type = _type
      ..disabled = _isAnchorLink ? null : props.isDisabled
      ..addProps(ariaProps()
        ..disabled = _isAnchorLink ? props.isDisabled : null
      )
    )(children);
  }

  ClassNameBuilder getButtonClasses() {
    return forwardingClassNameBuilder()
      ..add('btn')
      ..add('btn-block', props.isBlock)
      ..add('active', _isActive)
      ..add('disabled', props.isDisabled)
      ..add(props.skin.className);
  }
}
```

```dart
@Component(subtypeOf: ButtonComponent)
class ToggleButtonComponent 
    extends ButtonComponent<ToggleButtonProps, ToggleButtonState> {
  // ...

  // Instead of having to nest a full-blown Button() component within render, I can just 
  // override the root node classes, and call the root node rendering method to get 
  // the same result.

  @override
  render() {
    return renderButton(
      [
        renderInput(),
        props.children
      ]
    );
  }

  @override
  ClassNameBuilder getButtonClasses() {
    return super.getButtonClasses()
      ..add('btn--toggle');
  }
}
```

I've been meaning to bring it up with the team - since I feel it's definitely something we should consider for WSD, but I haven't gotten around to it.

Also - the component example above is not in this PR, its part of the stateful toggle button component demo I started working on, but haven't had time to finish yet.